lib/galaxy/web/base/controller.py

```diff
@@ -314,7 +314,9 @@ def _get_extended_config(self, trans):
             'toolbox'                       : trans.app.toolbox.to_dict(trans, in_panel=False),
             'toolbox_in_panel'              : trans.app.toolbox.to_dict(trans),
             'message_box_visible'           : trans.app.config.message_box_visible,
-            'show_inactivity_warning'       : trans.app.config.user_activation_on and trans.user and not trans.user.active
+            'show_inactivity_warning'       : trans.app.config.user_activation_on and trans.user and not trans.user.active,
+            'tool_shed_urls'                : trans.app.tool_shed_registry.tool_sheds.values() if trans.app.tool_shed_registry else [],
```

Comment:
```suggestion
            'tool_shed_urls'                : list(trans.app.tool_shed_registry.tool_sheds.values()) if trans.app.tool_shed_registry else [],
```

Otherwise this fails to serialize on python 3 because the map object can't be json encoded (https://github.com/galaxyproject/galaxy/blob/0fa287f76a675c49c2702681ac4404f050b77130/lib/galaxy/util/odict.py#L72). That'll go away fortunately with https://github.com/galaxyproject/galaxy/pull/8104 but until then that'll be necessary.